src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py

```diff
@@ -56,16 +59,16 @@ def load_arguments(self, _):
             c.argument('default_consistency_level', arg_type=get_enum_type(DefaultConsistencyLevel), help="default consistency level of the Cosmos DB database account")
             c.argument('max_staleness_prefix', type=int, help="when used with Bounded Staleness consistency, this value represents the number of stale requests tolerated. Accepted range for this value is 1 - 2,147,483,647")
             c.argument('max_interval', type=int, help="when used with Bounded Staleness consistency, this value represents the time amount of staleness (in seconds) tolerated. Accepted range for this value is 1 - 100")
-            c.argument('ip_range_filter', nargs='+', validator=validate_ip_range_filter, help="firewall support. Specifies the set of IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IPs for a given database account. IP addresses/ranges must be comma-separated and must not contain any spaces")
+            c.argument('ip_rules', nargs='+', help="firewall support. Specifies the set of IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IPs for a given database account. This should be an array of ipRanges or addresses", is_preview=True)
```

Comment:
I would suggest leaving the parameter name and format the same. Otherwise, users will have to update scripts whenever they update to the newest CLI. Internally, we can still just convert a comma-delimited string to the IP rules array.

If we want to keep the new name/array format, it might be helpful to provide an example of what an array might look like (ie is it space-separated or should they provide brackets?).

Either way, you can use the "validator" kwarg here (similar to validate_capabilities below). If you add it here, you will not need the "get_ip_rules" helper method in custom.py. All transformation from input format to required format can be done in the validator (similar to https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/cosmosdb/_validators.py#L38)